primeqa/mrc/processors/preprocessors/natural_questions.py

```diff
@@ -0,0 +1,233 @@
+                if example['context_token_to_char'][j] == -1:
+                    continue
+                if example['document_tokens']['start_byte'][j] >= example['target']['start_positions'][i]:
+                    break
+            if j < num_tokens:
+                example['target']['start_positions'][i] = example['context_token_to_char'][j]
+            else:
+                raise ValueError('Start position of short answer can not be set to the token based context.')
+
+            for j in range(num_tokens - 1, -1, -1):
+                if example['context_token_to_char'][j] == -1:
+                    continue
+                if example['document_tokens']['end_byte'][j] <= example['target']['end_positions'][i]:
+                    break
+            if j >= 0:
+                example['target']['end_positions'][i] = example['context_token_to_char'][j] + \
+                                                        len(example['document_tokens']['token'][j])
+            else:
+                raise ValueError('End position of short answer can not be set to the token based context.')
+
+        num_passages = len(example['passage_candidates']['start_positions'])
+        for i in range(num_passages):
+            passage_start_position = example['passage_candidates']['start_positions'][i]
+            passage_end_position = example['passage_candidates']['end_positions'][i]
+
+            for j in range(num_tokens):
+                if example['context_token_to_char'][j] == -1:
+                    continue
+                if example['document_tokens']['start_byte'][j] >= passage_start_position:
+                    break
+            if j < num_tokens:
+                example['passage_candidates']['start_positions'][i] = example['context_token_to_char'][j]
+            else:
+                raise ValueError('Start position of passage candidate can not be set to the token based context.')
+
+            for j in range(num_tokens - 1, -1, -1):
+                if example['context_token_to_char'][j] == -1:
+                    continue
+                if example['document_tokens']['end_byte'][j] <= passage_end_position:
+                    break
+            if j >= 0:
+                example['passage_candidates']['end_positions'][i] = example['context_token_to_char'][j] + \
+                                                                    len(example['document_tokens']['token'][j])
+            else:
+                raise ValueError('End position of passage candidate can not be set to the token based context.')
+
+        return example
+
+
+    def get_annotations(self, annotations, paragraphs):
+        """
+        Process NQ annotations into preprocessor format.
+        Args:
+             annotations: Annotations of NQ example.
+             paragraphs: Passage_candidates of NQ example.
+        Returns:
+             Annotations in preprocessor format.
+        """
+
+        tydi_annotations = {}
```

Comment:
Would be better to call this nq_annotations

Reply:
done